### PR TITLE
intel: Enable watchdog timer on Intel S10 platform

### PR DESCRIPTION
**Pull request by hadi-asyrafi**
Watchdog driver support & enablement during platform setup

Signed-off-by: Muhammad Hadi Asyrafi Abdul Halim <muhammad.hadi.asyrafi.abdul.halim@intel.com>

### CODE REVIEW

File: plat/intel/soc/stratix10/drivers/wdt/watchdog.c [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+ */
+
+#include <common/debug.h>
+#include <lib/mmio.h>
+#include <platform_def.h>
+
+#include "watchdog.h"
+
+
+// Reset watchdog timer
```
Comment (antonio-nino-diaz-arm) [inline plat/intel/soc/stratix10/drivers/wdt/watchdog.c]: Use ``/* */`` comments.

File: plat/intel/soc/stratix10/drivers/wdt/watchdog.c [hunk 2]
```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2019, ARM Limited and Contributors. All rights reserved.
```
Comment (antonio-nino-diaz-arm) [inline plat/intel/soc/stratix10/drivers/wdt/watchdog.c]: This is the old copyright header format, just say Intel.

File: plat/intel/soc/stratix10/drivers/wdt/watchdog.h [hunk 3]
```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2019, ARM Limited and Contributors. All rights reserved.
```
Comment (antonio-nino-diaz-arm) [inline plat/intel/soc/stratix10/drivers/wdt/watchdog.h]: Same two comments for this file.

File: plat/intel/soc/stratix10/drivers/wdt/watchdog.h [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+// Counter Restart Register
+#define WDT_SW_RST			(0x76)
+
+
+// Function declaration
+void watchdog_init(int watchdog_clk);
+void watchdog_enable(void);
+void watchdog_info(void);
+void watchdog_status(void);
+void watchdog_sw_rst(void);
```
Comment (antonio-nino-diaz-arm) [inline plat/intel/soc/stratix10/drivers/wdt/watchdog.h]: This header doesn't have header guards.

**Comment by antonio-nino-diaz-arm** [comment]
IP Review +1

**Comment by hadi-asyrafi** [comment]
Thanks for the comments. Fixed them up.

**Comment by antonio-nino-diaz-arm** [comment]
jenkins: test this please

**Comment by antonio-nino-diaz-arm** [comment]
The CI failure is not your fault, this PR is ok.

**Comment by antonio-nino-diaz-arm** [comment]
jenkins: test this please

**Comment by thloh85-intel** [comment]
Looks good to me 👍

**Pull request merged**